### PR TITLE
[FrameworkBundle][DX] Display a nice error message if an enabled component is missing

### PR DESCRIPTION
**Pull request by derrabus**
| Q             | A
| ------------- | ---
| Branch?       | 3.4
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #25093
| License       | MIT
| Doc PR        | N/A

### CODE REVIEW

**Comment by nicolas-grekas** [comment]
Could you please check if that happens for other components also?

File: src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php [hunk 1]
```diff
@@ -232,6 +232,10 @@ public function load(array $configs, ContainerBuilder $container)
         }
 
         if ($this->isConfigEnabled($container, $config['form'])) {
+            if (!class_exists('Symfony\Component\Form\Form')) {
```
Comment (nicolas-grekas) [inline src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php]: `Form::class`?

File: src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php [hunk 2]
```diff
@@ -232,6 +232,10 @@ public function load(array $configs, ContainerBuilder $container)
...
+                throw new LogicException('Form support cannot be enabled. Please install the Form component.');
```
Comment (nicolas-grekas) [inline src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php]: just to match similar messages: 
`Form support cannot be enabled as the Form component is not installed.`

**Review by nicolas-grekas — APPROVED** [review]

**Comment by xabbuh** [comment]
For other components we did this in 3.3. Should we merge the PR into the `3.3` as a bugfix?

**Comment by derrabus** [comment]
@nicolas-grekas Serializer is also affected. I can enable the component without having it installed and FrameworkBundle would register the `serializer` service, but it's impossible to wire it.

**Comment by derrabus** [comment]
I've also checked the validator: When enabling it, `FrameworkExtension` complains about the missing Translation component, which is not wrong, but seems a bit odd. However, if the Translation component is present, I get the expected exception about the missing Validator component.

File: src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php (hunk 2 above)
Comment (derrabus) [reply src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php]: ✅

**Comment by derrabus** [comment]
Do you think, we can remove those lines?

https://github.com/symfony/symfony/blob/4c930284acdde24946e1322f02c34c6b2b7cf631/src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php#L181-L183

Installing Validator without Translation is not possible, enabling validation without the Validator component will be caught elsewhere. So this block just generates a confusing error message.

**Review by nicolas-grekas — APPROVED** [review]
let's go for 3.3

**Comment by derrabus** [comment]
3.3 it is, then.

**Comment by nicolas-grekas** [comment]
Thank you @derrabus.

**Comment by nicolas-grekas** [comment]
@derrabus about https://github.com/symfony/symfony/pull/25094#issuecomment-346314120, you reverted your idea?

**Comment by derrabus** [comment]
@nicolas-grekas For now, yes. That change conflicted with the 3.3 branch. I will setup a 3.3 app tonight and test again.

**Pull request merged**